Source/WebKit/webpushd/WebPushDaemon.mm

```diff
@@ -330,6 +337,25 @@ void handleWebPushDMessage(ClientConnection* connection, Span<const uint8_t> enc
     function();
 }
 
+void Daemon::startIncomingPushTransactionTimer()
+{
+    if (!m_incomingPushTransaction)
+        m_incomingPushTransaction = adoptOSObject(os_transaction_create("com.apple.webkit.webpushd.daemon.incoming-push"));
+    m_incomingPushTransactionTimer.startOneShot(s_incomingPushTransactionTimeout);
+}
+
+void Daemon::cancelIncomingPushTransactionTimer()
+{
+    m_incomingPushTransactionTimer.stop();
+    m_incomingPushTransaction = nullptr;
+}
+
+void Daemon::incomingPushTransactionTimerFired()
+{
+    RELEASE_LOG_ERROR_IF(m_incomingPushTransaction, Push, "UI process failed to fetch push before incoming push transaction timed out.");
+    m_incomingPushTransaction = nullptr;
```

Comment:
Should we remove messages so that we do not jetsam?

Reply:
I think probably not, as the process might just be taking a long time to start up. If jetsam really needs to kill us, they should be able to since we are not dirty (no transactions held) and we are in a pretty low jetsam band already.